VideoLocker/src/test/java/org/edx/mobile/discussions/DiscussionTextUtilsTest.java

```diff
@@ -0,0 +1,216 @@
+package org.edx.mobile.discussions;
+
+import android.graphics.Typeface;
+import android.text.SpannableString;
+import android.text.TextUtils;
+import android.text.style.ForegroundColorSpan;
+import android.text.style.StyleSpan;
+import android.view.View;
+import android.widget.TextView;
+
+import org.edx.mobile.R;
+import org.edx.mobile.discussion.DiscussionTextUtils;
+import org.edx.mobile.discussion.IAuthorData;
+import org.edx.mobile.test.BaseTestCase;
+import org.edx.mobile.util.ResourceUtil;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.Date;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+
+public class DiscussionTextUtilsTest extends BaseTestCase {
+    @Test
+    public void testSetAuthorAttributionText_AllCombinations() {
+        final TextView textView = new TextView(context);
+
+        // Input values
+        final String author = "author";
+        final String label = "label";
+        final Date creationDate = new Date(1462368150195L); // A day before now
+
+        // Expected output constructs
+        final long now = 1462454550195L; // A day after the creationDate
+        final String relativeTime = "Yesterday";
```

Comment:
Yup thats out of scope.

Plus I don't think we need tests for these cases cuz we are using the SDKs provided function for relative time strings i.e. 

```java
DateUtils.getRelativeTimeSpanString(
                    timeMs,
                    nowMs,
                    DateUtils.SECOND_IN_MILLIS,
                    DateUtils.FORMAT_NUMERIC_DATE | DateUtils.FORMAT_SHOW_YEAR);
```